### PR TITLE
Fix serial disk add testcase logic

### PR DESCRIPTION
**Pull request by SRIKKANTH**
Current logic is adding and removing a single disk instead of adding 1 disk at a time till max supported disks.

Modified the logic to add 1 disk at a time till max allowed disks and for removal all attached disks one after the other.

### CODE REVIEW

**Comment by LiliDeng** [comment]
> Current logic is adding and removing a single disk instead of adding 1 disk at a time till max supported disks.
> 
> Modified the logic to add 1 disk at a time till max allowed disks and for removal all attached disks one after the other.

Is there any test issue for current code? Can you clarify the reason for changing code? Thanks.

**Comment by SRIKKANTH** [comment]
> > Current logic is adding and removing a single disk instead of adding 1 disk at a time till max supported disks.
> > Modified the logic to add 1 disk at a time till max allowed disks and for removal all attached disks one after the other.
> 
> Is there any test issue for current code? Can you clarify the reason for changing code? Thanks.

Current testcase simply verifies adding removing a single disk instead of actually checking if its allowed to attach one disk at a time till max supported disks.

Is this expected behavior? 

If yes why the add remove iteration count is equal to max disks allowed on the SKU why not add remove with a fixed number of times. 

I think we should validate adding disks serially till max allowed disks and remove all the attached disks in the end.

**Comment by adityagesh** [comment]
Your change aligns with the description of the test case (pasting it below)

```
        1. Get maximum number of data disk for the current vm_size.
        2. Get the number of data disks already added to the vm.
        3. Serially add and remove the data disks and verify that the added
        disks are present in the vm.
```

However, the name of the function is misleading, how about renaming it similar to something like hot_add_disk_serial_one_by_one? Similar to how we have for the nic

Thoughts on this? @LiliDeng

File: microsoft/testsuites/core/storage.py [hunk 1]
```diff
@@ -567,11 +567,14 @@ def _hot_add_disk_serial(
 
         # get partition info before adding data disk
         partitions_before_adding_disk = lsblk.get_disks(force_run=True)
-
+        added_disk_count = 0
+        disks_added = []
         for _ in range(disks_to_add):
             # add data disk
             log.debug("Adding 1 managed disk")
```
Comment (LiliDeng) [inline microsoft/testsuites/core/storage.py]: remove this log

Comment (SRIKKANTH) [reply microsoft/testsuites/core/storage.py]: Removed.

**Review by LiliDeng — APPROVED** [review]

**Pull request merged**